Source/Android/jni/MainAndroid.cpp

```diff
@@ -144,6 +145,10 @@ void Host_YieldToUI()
 
 void Host_TitleChanged()
 {
+  s_game_metadata_is_valid = true;
+
+  JNIEnv* env = IDCache::GetEnvForThread();
```

Comment:
This line seems to be the cause of Gecko OS crashing. Removing it and the associated line below allows Gecko OS to load mods seemingly without problem. Only removing the associated line below didn't help.

Reply:
Huh. Guess there's a reason why `Host_RequestRenderWindowSize` is using a temporary thread, then... I'm still not quite sure what's wrong, but this is a good hint.

Reply:
I've made this use the same workaround as `Host_RequestRenderWindowSize`. Please test it.